ci/conda_env_python.yml

```diff
@@ -23,3 +23,9 @@ pytest
 python
 setuptools
 setuptools_scm
+# Requirements for building the documentation
+breathe
+ipython
+recommonmark
```

Comment:
What is recommonmark?

Reply:
`conda_env_cpp.yml` and `conda_env_python.yml` files contain arrow-cpp's and pyarrow's dependencies and eventually will be used to spin up the development environment, see https://github.com/apache/arrow/pull/2836

I suggest to create a `ci/conda_env_sphinx.yml` or `conda_env_doc.yml` with the sphinx dependencies, including `doxygen` and `rsync` too.